packages/data-transformer/src/transformers/data.transformer.spec.ts

```diff
@@ -0,0 +1,143 @@
+import {DataTransformerBuilder} from "./data-transformer.builder";
+import {DataTransformerInterceptor} from "../interfaces/data-transformer-interceptor.interface";
+import {DataTransformerInterceptorUniqueKeyType} from "../types/data-transformer-interceptor-unique-key.type";
+import {DataTransformerRow} from "../types/data-transformer.row";
+import 'jest-extended';
+
+describe('Data Transformer', () => {
+    it("should properly transform", async () => {
+        const dataTransformer = new DataTransformer([new LowercaseNormalizer()], []);
+
+        const source = [{
+            NAME: "Etienne Noel",
+            province: "QUEBEC",
+            TOTAL: 10,
+        }];
+
+        const dataTransformerBuilder = new DataTransformerBuilder();
+        dataTransformerBuilder
+            .add()
+                .setSourceProperty("NAME")
+                .setDestinationProperty("name")
+                .end()
+            .add()
+                .setSourceProperty("province")
+                .addNormalizer(LowercaseNormalizer.name)
+                .setDestinationProperty("province")
+                .end()
+            .add()
+                .setSourceProperty("TOTAL")
+                .setDestinationProperty("total")
+                .end();
+
+        const destination = await dataTransformer.transform(dataTransformerBuilder, source);
+
+        expect(destination.length).toBe(1)
+
+        expect(destination[0].name).toBeDefined()
+        expect(destination[0].name).toBe("Etienne Noel");
+        expect(destination[0].province).toBe("quebec");
+        expect(destination[0].total).toBe(10);
+    })
+
+    it("should properly transform an array with numerical indices", async () => {
+        const dataTransformer = new DataTransformer([new LowercaseNormalizer()], []);
+
+        const source = [
+            ["Etienne Noel", "QUEBEC", 10],
+        ];
+
+        const dataTransformerBuilder = new DataTransformerBuilder();
+        dataTransformerBuilder
+            .add()
+                .setSourceProperty("0")
+                .setDestinationProperty("name")
+                .end()
+            .add()
+                .setSourceProperty("1")
+                .addNormalizer(LowercaseNormalizer.name)
+                .setDestinationProperty("province")
+                .end()
+            .add()
+                .setSourceProperty("2")
+                .setDestinationProperty("total")
+                .end();
+
+        const destination = await dataTransformer.transform(dataTransformerBuilder, source);
+
+        expect(destination.length).toBe(1)
+
+        expect(destination[0].name).toBeDefined()
+        expect(destination[0].name).toBe("Etienne Noel");
+        expect(destination[0].province).toBe("quebec");
+        expect(destination[0].total).toBe(10);
+    })
+
+    it("should properly call the before row transformers and respect the order of calls", async () => {
```

Comment:
Could you add a test where the before and after row actually merged properties to validate it works as expected